cmd/thanos/receive.go

```diff
@@ -131,16 +129,22 @@ func registerReceive(app *extkingpin.App) {
 			AllowOverlappingBlocks: *tsdbAllowOverlappingBlocks,
 		}
 
-		// Local is empty, so try to generate a local endpoint
-		// based on the hostname and the listening port.
+		// Make sure the both localEndpoint and (hashringsFilePath/hashringsFileContent) are not empty
```

Comment:
This seems wrong to me as it makes it impossible for a node to be in "ingestor" mode. It's absolutely fine for a Receive node to have no local endpoint and no hashring configured. This is the default way the receiver runs: all data it receives is deemed to be for itself and is stored in a local TSDB.

Reply:
I agree with your thoughts, we currently try to fetch the host as our local endpoint, so this behavior has removed that functionality. 

Originally, I thought that we should explicitly pass the `localEndpoint`, but now with the [other approach](https://github.com/thanos-io/thanos/pull/4231#discussion_r632469132), we don't have to change the behavior as such, and I can totally remove this check!